test/integration/npm/publish.test.js

```diff
@@ -51,12 +51,38 @@ describe('npm publish', function () {
       }));
   });
 
-  it('should trigger a carpenter build', function (done) {
+  it('should trigger a carpenter build with promote=true', function (done) {
```

Comment:
You should have 3 tests:

1. `"wrhs":{"autoPromoteOnPublish":false}`
1. `"wrhs":{"autoPromoteOnPublish":true}`
1. no config

Reply:
@msluther fair, will add the 2nd